base/util.go

```diff
@@ -873,6 +873,20 @@ func Crc32cHashString(input []byte) string {
 	return fmt.Sprintf("0x%x", Crc32cHash(input))
 }
 
+func SplitHostPort(hostport string) (string, string, error) {
+	host, port, err := net.SplitHostPort(hostport)
+	if err != nil {
+		return "", "", err
+	}
+
+	// If this is an IPv6 address, we need to rewrap it in []
```

Comment:
This comment is really useful, but if there's any more details as to why this is needed .. would be even better.  Is it something gocb needs?  Or is it the standard way to represent ipv6 addresses as a string?

Reply:
This is standard ipv6 address formatting.  (and also based on prior art in gocb)

Reply:
Ok